### DISCUSSION TITLE
splitdns-magic-with-tailscale/

### DISCUSSION THREAD

**Discussion by giscus** [ironicbadger/blog.ktz.me#222] [Announcements]
# splitdns-magic-with-tailscale/

Today, we're going to take a deep dive into the world of DNS. Specifically looking at Tailscale's magicDNS feature which allows us to do neat things like refer to our Tailnet devices by name or configure splitDNS to query remote DNS servers.

https://blog.ktz.me/splitdns-magic-with-tailscale/

**Comment by Janov911** [comment]
great piece! And it works "as advertised" 
Do you happen to know how I can tie my pihole into this? At home I use my pihole as my DNS, and wiregurd into my network when I'm on the road.
I heard you and Chris (on selfhosted) talk about removing the need for wireguard, but have not been able to make this work.
As soon as I set the pihole as the DDNS for the tailscale network all name resolution goes out the window. (and yes, ip forwarding is on, all origins is ticked and I can ping the box via it's ip)

Maybe an idea for another tutorial?

**Comment by pofo14** [comment]
Is there a tutorial you have for how to configure the 5 layer deep domain naming on the servers on the network

**Comment by t4ir1** [comment]
Hi, great article.
Unfortunately this is not working for me. I read on a reddit post that instead of advertising a routes and exposing your LAN to your tailnet, one could simply configure the splitDNS with the tailnet IP of the DNS server. Unfortunately this workaround also doesn't work for me.
I'm one of those sad engineers that has to deal with Windows and Windows Server DNS, I wonder if this is the problem I am facing although the windows firewall seems to be clean in this regard.
Resolving DNS on LAN works perfectly so I am having a hard-time understanding why is split DNS not working for me.
Any help would be greatly appreciated!
